src/Idris/Package.idr

```diff
@@ -360,6 +360,23 @@ install pkg opts -- not used but might be in the future
          coreLift $ changeDir srcdir
          runScript (postinstall pkg)
 
+-- Check package without compiling anything.
+check : {auto c : Ref Ctxt Defs} ->
+        {auto s : Ref Syn SyntaxInfo} ->
+        {auto o : Ref ROpts REPLOpts} ->
+        PkgDesc ->
+        List CLOpt ->
+        Core (List Error)
+check pkg opts =
+  do
+    defs <- get Ctxt
+    addDeps pkg
+    processOptions (options pkg)
+    let toBuild = maybe (map snd (modules pkg))
+                        (\m => snd m :: map snd (modules pkg))
+                        (mainmod pkg)
+    buildAll toBuild
```

Comment:
Can we refactor `build` to reuse this shared prefix? Also, looking at `build`,
it seems you are missing a call to [`preOptions`](https://github.com/idris-lang/Idris2/blob/71db21b19d42acff8ce53b8ee036287dcc67afc1/src/Idris/SetOptions.idr#L36):

> Options to be processed before type checking. Return whether to continue.

Reply:
Should we run also `preBuild`/`postBuild` scripts during `check`?